src/libs/API.js

```diff
@@ -105,7 +105,11 @@ function makeRequestWithSideEffects(command, apiCommandParameters = {}, onyxData
  * @param {Object} [onyxData.failureData] - Onyx instructions that will be passed to Onyx.update() when the response has jsonCode !== 200.
  */
 function read(command, apiCommandParameters, onyxData) {
-    makeRequestWithSideEffects(command, apiCommandParameters, onyxData, CONST.API_REQUEST_TYPE.READ);
+    // We need to ensure all the write requests on the sequential queue have responded first before running read requests.
+    // This is because if we make write requests offline, their optimisticData can get overwritten by responses from
+    // lightweight read requests that access the same Onyx key and modify its value before the write request has had a chance to update it after responding.
+    // This prevents the "flickering" effect.
```

Comment:
This comment is a good start, but I've got some ideas to make it even better:

- Remove the line about 'flickering effect'. That is ambiguous and not a real term so no one is going to know what it means. I suggest either removing the line completely or rewording it to explain what really happens.
- Remove the mention of "lightweight" because network requests are data, they don't have any weight
- Replace all pronouns (eg. their, it, its) with the references to what they actually are

Reply:
This is really helpful feedback, thanks! I always have go through like 5 different versions of these types of comments

Reply:
Just pushed a new comment, please let me know what you think!
```
    // Ensure all write requests on the sequential queue have finished responding before running read requests.
    // Responses from read requests can overwrite the optimistic data inserted by
    // write requests that use the same Onyx keys and haven't responded yet.
```